calculator.rb

```diff
@@ -0,0 +1,96 @@
+# Baseline
+# The program should ask the user for an operation (as a string symbol) and two numbers.
+#
+# Primary Requirements
+# The program should use the input operation and two numbers to provide the result of applying the operation to the two numbers.
+#
+# The program should have support for these four operations: addition, subtraction, multiplication, and division.
+# The program should accept both the name (add) and the symbol (+) for each possible operation.
+# The program should handle unexpected user input
+
+
+# Creating a simple method to explain each of the mathematical operators.
+def explanation ()
+  puts "\nBASIC MATHEMATICAL OPERATORS AND THEIR DEFINITIONS:"
+  puts "1. Add (+) - addition, summation, plus, increase, total."
+  puts "2. Subtract (-) - taking away, minus, less, difference, decrease, deduct."
+  puts "3. Multiply (*) - aka multiplication, the product of, times."
+  puts "4. Divide (/) - aka division, quotient, how many times goes into."
+end
+
+#Getting the user input on which type of mathematical operator they'd like to use.
+puts "Welcome to Carly's Calculator."
+explanation
+print "\nWhich of the above mathematical operators would you like to choose:  Add(+), Subtract(-), Multiply(*), or Divide(/)?  "
+operator = gets.chomp.downcase
+
+#Allowing the user to choose their mathematical operator, and accounting for the possibility that they don't choose a valid option.  Offering a refresh of the tutorial.
+until ["add", "subtract", "multiply", "divide", "+", "-", "*", "/"].include?(operator)
+  print "That is not one of the valid mathematical operators!  Would you like to view a full explanation of the operators again? (yes or no)  "
+  want_explanation = gets.chomp.downcase
+    if want_explanation == "yes" || want_explanation == "sure"
+      explanation
+    end
```

Comment:
just me being nitpicky-- this `if ... end` block is indented one indent too much!